packages/docusaurus-plugin-content-docs/src/sidebarItemsGenerator.ts

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  SidebarItem,
+  SidebarItemDoc,
+  SidebarItemCategory,
+  SidebarItemsGenerator,
+  SidebarItemsGeneratorDoc,
+} from './types';
+import {sortBy, take, last, orderBy} from 'lodash';
+import {addTrailingSlash, posixPath} from '@docusaurus/utils';
+import {Joi} from '@docusaurus/utils-validation';
+import {extractNumberPrefix} from './numberPrefix';
+import chalk from 'chalk';
+import path from 'path';
+import fs from 'fs-extra';
+import Yaml from 'js-yaml';
+import {DefaultCategoryCollapsedValue} from './sidebars';
+
+const BreadcrumbSeparator = '/';
+
+export const CategoryMetadataFilenameBase = '_category_';
+export const CategoryMetadataFilenamePattern = '_category_.{json,yml,yaml}';
+
+export type CategoryMetadatasFile = {
+  label?: string;
+  position?: number;
+  collapsed?: boolean;
+
+  // TODO should we allow "items" here? how would this work? would an "autogenerated" type be allowed?
```

Comment:
@slorber Assuming based on this that I won't be able to do what I asked about in an earlier comment?
https://github.com/facebook/docusaurus/issues/3464#issuecomment-966271746

Reply:
No I don't think it's possible atm but we may add support for it if there's a good use-case exposed clearly, preferably with a concrete example showcasing the problem (seeing sources is better)